nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Utilities/MSBuildHelper.cs

```diff
@@ -211,6 +198,58 @@ public static IEnumerable<Dependency> GetTopLevelPackageDependenyInfos(Immutable
         }
     }
 
+    /// <summary>
+    /// Given an MSBuild string and a set of properties, returns our best guess at the final value MSBuild will evaluate to.
+    /// </summary>
+    /// <param name="msbuildString"></param>
+    /// <param name="propertyInfo"></param>
+    /// <returns></returns>
+    public static string GetRootedValue(string msbuildString, Dictionary<string, string> propertyInfo)
+    {
+        bool modified;
+        var seenProperties = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+        do
+        {
+            modified = false;
+            if (TryGetPropertyName(msbuildString, out var propertyName))
```

Comment:
This can now become a while loop and we can remove the `modified` variable.
```suggestion
        while (TryGetPropertyName(msbuildString, out var propertyName))
        {
```